File: rust/src/hello_world.rs
// Hello World in Rust !

use chrono::{DateTime, Local};
use std::env;

fn print_greeting(args: &[String]) {
  if args.len() > 1 {
    println!("Hi {}, greetings from Rust !", args[1..].join(" "));
  } else {
    println!("Hello World from Rust !");
  }
}

fn print_date() {
  let now: DateTime<Local> = Local::now();
  println!("Today is {}", now.format("%B %e, %Y %H:%M:%S"));
}

fn main() {
  let args: Vec<_> = env::args().collect();
  print_greeting(&args);
  print_date();
}

File: dotnet/src/HelloWorld.cs
// Hello World in Dotnet !

namespace Hello.World;

public class HelloWorld {
    public static void PrintGreeting(string[] args) {
        if (args.Length > 0) {
            string name = string.Join(" ", args);
            Console.WriteLine($"Hi {name}, greetings from Dotnet !");
        } else {
            Console.WriteLine("Hello World from Dotnet !");
        }
    }

    public static void PrintDate() {
        DateTime currentDate = DateTime.Now;
        string layout = "MMMM d, yyyy HH:mm";
        string formattedDate = currentDate.ToString(layout);
        Console.WriteLine($"Today is {formattedDate}");
    }

    static void Main(string[] args) {
        PrintGreeting(args);
        PrintDate();
    }
}
